### ISSUE TITLE
How is the web optimized for Canvas? (CustomPaint / Skia CanvasKit)

### ISSUE THREAD

**Issue by softmarshmallow** [flutter/gallery#353]
https://gallery.flutter.dev/#/demo/2d-transformations

As we can see, Canvas performance on web is super slow. I don't understand this performance. It's basically running on skia canvaskit. How could it get this slow?

Is this the best performance we can get via the flutter canvas on web?

Providing flutter web canvas preformance example will be great for many developers to choose flutter for their graphic-related service.

**Comment by clocksmith** [comment]
@justinmc Can you take a look?

**Comment by justinmc** [comment]
I've wondered this myself, but I worry that it may be partially due to my own poor use of CustomPaint when I wrote the demo.  CC @nturgut in case you have any observations off the top of your head about why this demo would be so slow.

I've noticed that the demo's performance is fine if I do any of the following:

  * Replace the CustomPaint with a Container.
  * Change the CustomPaint to draw a single rect.
  * Change the CustomPaint to draw a single triangle with drawVertices.

However, when using the single CustomPaint to draw many (100-1000) triangles, the framerate is very slow when interacting.  All other platforms do not have this problem.

I plan to come back to this at some point and see if I can speed things up.  Maybe web should be as fast as the other platforms here, but I also think I may be doing things to make it especially slow.
